pkg/apiserver/handlers.go

```diff
@@ -43,6 +42,11 @@ var specialVerbs = map[string]bool{
 	"watch":    true,
 }
 
+// namespaceSubresouces is a set of all the subresources available on a namespace resource.  This is a special case because
+// URLs look like api/v1beta3/namspaces/<namespace name>/[subresource | resource].  We need to be able to distinguish the two
+// different cases.
+var namespaceSubresources = util.NewStringSet("status", "finalize")
```

Comment:
Also a TODO on using the swagger model to test the APIRequestInfo and match up behavior to responses.